lib/src/ruby/ruby.cc

```diff
@@ -47,6 +50,13 @@ namespace facter { namespace ruby {
 
     void load_custom_facts(collection& facts, bool initialize_puppet, vector<string> const& paths)
     {
+#ifdef _WIN32
+        // Initialize WSA before resolving custom facts. The Ruby runtime does this only when running
+        // in a Ruby process, it leaves it up to us when embedding it. See
+        // https://github.com/ruby/ruby/blob/v2_1_9/ruby.c#L2011-L2022 for comments.
+        // The only piece we seem to need out of rb_w32_sysinit is WSAStartup.
+        util::windows::wsa winsocket;
```

Comment:
When the object is destroyed, it'll cleanup the WSA config. load_custom_facts definitely doesn't get called from inside its own execution.